partitionKey in the cookies API

rebloor opened the pull request:
#### Summary
Added details about the inclusion of partitionKey in the cookies API to the:
- the cookies object
- the Firefox 94 release notes

#### Related issues
Addresses  [Bug 1669716](https://bugzilla.mozilla.org/show_bug.cgi?id=1669716) (https://github.com/mdn/content/issues/9363)

#### Metadata

- [ ] Adds a new document
- [ ] Rewrites (or significantly expands) a document
- [x] Fixes a typo, bug, or other error

rebloor committed:
Added details about the inclusion of partitionKey in the cookies API to:
- the cookies object
- the Firefox 94 release notes

Rob--W reviewed (changes requested):
Property also needs to be documented at:
https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/cookies/get
https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/cookies/getAll
https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/cookies/set
https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/cookies/remove

The feature needs to prominently be documented at the main page too. For example, see the level of depth in the documentation of a similar property: https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/cookies#first-party_isolation

The section should include:
- Reference to https://developer.mozilla.org/en-US/docs/Web/Privacy/State_Partitioning
- While this feature is internally called "dynamic first-party isolation (dFPI)", its public name is "dynamic partitioning" - https://developer.mozilla.org/en-US/docs/Web/Privacy/State_Partitioning#dynamic_state_partitioning
- Partitioned cookies are not visible to getAll by default. This is an intentional design decision, because extensions that aren't aware of partitioned cookies may inadvertently copy cookies to another partition. For a write-up of examples, see https://phabricator.services.mozilla.com/D123491#4106552
- From the documentation, it should be obvious that extensions that want to do the right thing should be aware of `partitionKey`. To maintain compatibility with browsers that don't support `partitionKey`, extensions can check for the presence of `partitionKey` on the returned cookie, and if it is set, put it on the object of the cookie that they're updating or removing (some extensions that want to update cookies would use `cookies.get` to get a cookie, and then use `cookies.set` with the details of the cookie).

Let me know if you have more questions about this API or the level of documentation.

rebloor committed:
Added partitionKey to cookies.get, .getAll, .set,  and .remove

rebloor committed:
Additional information about prostitutionKey on the cookies API and release notes

rebloor committed:
Updated tracking prevention information in the cookies API overview and clarified dynamic partitioning in the state partitioning page. Also annotated partitionKey as optional in get, getAll, set, and remove.

Trikolon reviewed (changes requested):
Feedback for the state partitioning article changes. Thank you for working on this!

rebloor committed:
Addressed feedback regarding status information for dynamic partitioning and formatting of new content

rebloor commented:
@Trikolon Thanks for your review, the re-review button doesn't work for me, could you take a look at my changes.

Trikolon reviewed (approved):
Thanks!

rebloor committed:
WIP changes for feedback

Rob--W reviewed (commented):
A few more small remarks (on top of the changes requested during my last review).

I found those after looking at the preview from the github-actions bot.

rebloor committed:
Various updates for feedback

Rob--W reviewed (approved):
The Github actions bot failed to update the preview, but it seems to look good enough. Thanks!

rebloor committed:
Changes to address feedback

rebloor committed:
Clarification regarding partitionKey

Co-authored-by: Rob Wu <rob@robwu.nl>

rebloor committed:
Added cross-references from methods and objects to dynamic partitioning section in cookies API overview

Rob--W reviewed (commented):
Just one more revision, after that rebase the PR so that it can be merged.

rebloor committed:
Change to use "dynamic partitioning"

rebloor committed:
Reset the terminology to use "storage partitioning" etc.

Rob--W reviewed (approved):
One more overlooked change. Please add the suggested change and rebase.

Then the bot should be able to generate a preview.

If it looks good to you, merge it \o/

rebloor committed:
Link correction

Co-authored-by: Rob Wu <rob@robwu.nl>

Rob--W reviewed (approved):
Changes still LGTM. Unfortunately, it seems that there are now many merge conflicts, likely due to #9910. Feel free to merge after resolving the merge conflict. If you'd like another review, that's also fine by me.

rebloor committed:
Resolve conflicts caused by preparation and conversion to markdown in

#	files/en-us/mozilla/add-ons/webextensions/api/cookies/index.html
#	files/en-us/mozilla/add-ons/webextensions/api/cookies/get/index.html
#	files/en-us/mozilla/add-ons/webextensions/api/cookies/getall/index.html
#	files/en-us/mozilla/add-ons/webextensions/api/cookies/remove/index.html
#	files/en-us/mozilla/add-ons/webextensions/api/cookies/set/index.html
#	files/en-us/web/privacy/state_partitioning/index.html

rebloor committed:
Removing state_partitioning/index.html (as it has been replaced by index.md)
